apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/017-add-use-case-form-and-example-org.migration.ts

```diff
@@ -0,0 +1,153 @@
+import { isSelfHostedInstance } from "@local/hash-isomorphic-utils/instance";
+import {
+  systemEntityTypes,
+  systemPropertyTypes,
+} from "@local/hash-isomorphic-utils/ontology-type-ids";
+import type { BaseUrl } from "@local/hash-subgraph";
+
+import {
+  createOrg,
+  getOrgByShortname,
+} from "../../../knowledge/system-types/org";
+import type { MigrationFunction } from "../types";
+import {
+  anyUserInstantiator,
+  createSystemEntityTypeIfNotExists,
+  createSystemPropertyTypeIfNotExists,
+} from "../util";
+
+const migrate: MigrationFunction = async ({
+  context,
+  authentication,
+  migrationState,
+}) => {
+  if (isSelfHostedInstance) {
+    /**
+     * Functionality is only relevant to hosted HASH, i.e. the instance at https://[app].hash.ai
+     */
+    return migrationState;
+  }
+
+  /**
+   * Create the entity type to hold information about a user's potential use(s) of HASH
+   */
+
+  const rolePropertyType = await createSystemPropertyTypeIfNotExists(
+    context,
+    authentication,
+    {
+      propertyTypeDefinition: {
+        title: "Role",
+        description: "The name of someone or something's role.",
+        possibleValues: [{ primitiveDataType: "text" }],
+      },
+      webShortname: "hash",
+      migrationState,
+    },
+  );
+
+  const intendedUsePropertyType = await createSystemPropertyTypeIfNotExists(
+    context,
+    authentication,
+    {
+      propertyTypeDefinition: {
+        title: "Intended Use",
+        description:
+          "The name or description of someone's intended use of something",
+        possibleValues: [{ primitiveDataType: "text" }],
+      },
+      webShortname: "hash",
+      migrationState,
+    },
+  );
+
+  const currentApproachPropertyType = await createSystemPropertyTypeIfNotExists(
+    context,
+    authentication,
+    {
+      propertyTypeDefinition: {
+        title: "Current Approach",
+        description:
+          "The name or description of the current approach to something",
+        possibleValues: [{ primitiveDataType: "text" }],
+      },
+      webShortname: "hash",
+      migrationState,
+    },
+  );
+
+  const willingToPayPropertyType = await createSystemPropertyTypeIfNotExists(
+    context,
+    authentication,
+    {
+      propertyTypeDefinition: {
+        title: "Willing To Pay",
+        description: "The amount that someone is willing to pay for something",
+        possibleValues: [{ primitiveDataType: "text" }],
+      },
+      webShortname: "hash",
+      migrationState,
+    },
+  );
+
+  const _prospectiveUserDefinitionEntityType =
```

Comment:
Just curious, why are we not attaching this information directly to the `User` entity? I assume we will need to use the `ownedById` of the entity to determine which account this "Prospective User" is associated with, so I suppose this is not a big deal.

Reply:
I figured these are temporary things and so didn't want to create further unnecessary types / links... we can always add them later if we want